delta/plugins/project-deletion/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/projectdeletion/ProjectDeletionPlugin.scala

```diff
@@ -0,0 +1,97 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.projectdeletion
+
+import ch.epfl.bluebrain.nexus.delta.plugins.projectdeletion.ProjectDeleter.projectDeletionPass
+import ch.epfl.bluebrain.nexus.delta.plugins.projectdeletion.model.ProjectDeletionConfig
+import ch.epfl.bluebrain.nexus.delta.sdk.ProjectResource
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.ProjectSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.plugin.Plugin
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.{Projects, ProjectsStatistics}
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sourcing.stream.{CompiledProjection, ExecutionStrategy, ProjectionMetadata, Supervisor}
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.{Task, UIO}
+
+import java.time.Instant
+import scala.concurrent.duration.DurationInt
+
+/**
+  * ProjectDeletion process reference, also the plugin instance that allows for graceful stop. Once the instance is
+  * constructed, the process would be already running.
+  *
+  * @param doStop
+  *   function which stops the project deletion stream
+  */
+class ProjectDeletionPlugin private (doStop: Task[Unit]) extends Plugin {
+
+  private val logger: Logger = Logger[ProjectDeletionPlugin]
+  logger.info("Project Deletion periodic check has started.")
+
+  override def stop(): Task[Unit] = {
+    doStop.timeout(10.seconds).flatMap {
+      case Some(()) => UIO.delay(logger.info("Project Deletion plugin stopped."))
+      case None     => UIO.delay(logger.error("Project Deletion plugin could not be stopped in 10 seconds."))
+    }
+  }
+}
+
+object ProjectDeletionPlugin {
+
+  private val logger: Logger = Logger[ProjectDeletionPlugin]
+
+  private val projectionMetadata: ProjectionMetadata = ProjectionMetadata("system", "project-deletion", None, None)
+
+  /**
+    * Constructs a stoppable ProjectDeletion process that is started in the background.
+    */
+  def started(
```

Comment:
I think we need to talk about this, it doesn't make sense to me for this logic to be in the deleter, as the functionality is different. This feels like the responsibility of the plugin rather than the deletion logic